pkg/util/machines.go

```diff
@@ -258,3 +258,41 @@ func MachinesAsString(machines []*clusterv1.Machine) string {
 	}
 	return message
 }
+
+// IsInternalIP checks if the IP is private/internal
+func IsInternalIP(ip string) bool {
+	netIP := net.ParseIP(ip)
+	if netIP == nil {
+		return false
+	}
+
+	privateBlocks := []*net.IPNet{
+		{IP: net.IPv4(10, 0, 0, 0), Mask: net.CIDRMask(8, 32)},     // 10.0.0.0/8
+		{IP: net.IPv4(172, 16, 0, 0), Mask: net.CIDRMask(12, 32)},  // 172.16.0.0/12
+		{IP: net.IPv4(192, 168, 0, 0), Mask: net.CIDRMask(16, 32)}, // 192.168.0.0/16
+		{IP: net.IPv4(127, 0, 0, 1), Mask: net.CIDRMask(8, 32)},    // 127.0.0.0/8 (loopback)
+		{IP: net.IPv4(169, 254, 0, 0), Mask: net.CIDRMask(16, 32)}, // 169.254.0.0/16 (link-local)
+	}
```

Comment:
note that cluster local subnets are not the same as local network subnets.

Reply:
i don't think that would be the correct way to determine if the ip is internal. it would be an internal network ip, but not necessary the internal cluster ip allocated to a node. to check if the ip is internal as per k8s' definition, i think one would have to check how the controller manager cidr is configured.